cmd/clusterctl/pkg/client/cluster/move.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// objectReference defines a reference to a Kubernetes object that is visited during the discovery phase for the move operation.
+type objectReference struct {
```

Comment:
Originally I was using metav1.ObjectReference , but then it was changed following https://github.com/kubernetes-sigs/cluster-api/pull/2060#discussion_r366572589

Reply:
Unless I'm mistaken, the comment was more because our local `objectReference` struct was embedding only a single field. I was suggesting to use metav1.ObjectReference directly, without embedding. @ncdc wdyt?

Reply:
Local ObjectReference was embedding objectReference + namespace string

Reply:
I see

Reply:
It was using **Owner**Reference originally. I'm fine with ObjectReference.

Reply:
@fabriziopandini do you want to change this or deal with a possible change in a follow-up PR?

Reply:
@ncdc if it is not a problem for you, I will address this into https://github.com/kubernetes-sigs/cluster-api/pull/2102

Reply:
SGTM, thanks!